Add task to get observations from GEOS location

mranst opened the pull request:
## Description

#575 This PR adds a task to copy all files from `/discover/nobackup/projects/gmao/dadev/rtodling/archive/542/prePP/ioda/geos_atmosphere/<datetime>`. I called this `GetGeosJediBackground`, though I can change this if there's a more ideal name. This task requires an entry in `experiment.yaml`:
```yaml
path_to_geos_jedi_background: /discover/nobackup/projects/gmao/dadev/rtodling/archive/542/prePP/ioda
```
It can then be called using:
`swell task GetGeosJediBackground experiment.yaml -m geos_atmosphere -d 20250321T120000Z`.

I didn't add it to any suite's flow.cylc yet, but the machinery is there to add that entry when the task is called from by a workflow

mranst committed:
Initial commit

rtodling commented:
@mranst @ashiklom I think there is a misunderstanding here. What I requested has nothing to do with getting the background. 

The directory I pointed to:  /discover/nobackup/projects/gmao/dadev/rtodling/archive/542/prePP/ioda

has observation files - not background files. The request relates to the GetObservations task.

rtodling reviewed (dismissed):
Unfortunately, all the changes here are not correct.  All these have to be wiped out and changes need to be made around - but not in the same task - associated w/ GetObservations.

rtodling commented:
BTW This task needs to be re-titled: Add task to get observations from GEOS location

rtodling commented:
The above would have to look something like this:



ioda_locations_not_in_r2d2: /discover/nobackup/projects/gmao/dadev/rtodling/archive/542/prePP/ioda

mranst commented:
@rtodling Is the behavior of this task what you were expecting (copying all files from that directory into the cycle directory), and the naming is the issue? Or does this need to be completely scrapped and incorporated into a different/existing task (GetObservations)

rtodling commented:
> @rtodling Is the behavior of this task what you were expecting (copying all files from that directory into the cycle directory), and the naming is the issue? Or does this need to be completely scrapped and incorporated into a different/existing task (GetObservations)

@mranst The behavior is correct, but it needs to be scrapped and placed in the GetObs task - getting observations has nothing to do with/ background - you should not need and have any changes in anything associated with/ the backgrounds - the idea is to have a opt within GetObs that bypasses all that GetObs is doing and simply copying all from the given directory to the run directory.

mranst committed:
intermediate

mranst committed:
rename task and add toggle

mranst committed:
restore get_observations

mranst committed:
fix toggle

mranst committed:
Updated to link

mranst commented:
I've updated the language to reflect that these are existing observations, and added a question trigger to the `flow.cylc` of  `3dfgat_atmos` as an example, I can remove/rework it depending on how we want to use this task. Files are now linked instead of copied

mranst committed:
Merge branch 'develop' into feature/mranst/get_geos_jedi_background

Dooruk commented:
@rtodling, are the observations in this folder going to be used and do they change? @ftgoktas is looking into registering more atmosphere observations into R2D2. So if you standardize the locations we could register files with experiment name as an index (542 or x0051 whichever).

however, I just noticed you separated some observation files into variables. I guess that is to resolve the bug regarding aircraft observations?

rtodling commented:
> @rtodling, are the observations in this folder going to be used and do they change? @ftgoktas is looking into registering more atmosphere observations into R2D2. So if you standardize the locations we could register files with experiment name as an index (542 or x0051 whichever).
> 
> however, I just noticed you separated some observation files into variables. I guess that is to resolve the bug regarding aircraft observations?

@Dooruk I'd strongly advocate against creating an r2d2 database based on the observations I generated - since things are still changing on my end ... that's way having this PR going is important, since it should not care of what's in the database - simply link/copy all ... I am going to pull this branch today and see how it works for me.

rtodling commented:
> I've updated the language to reflect that these are existing observations, and added a question trigger to the `flow.cylc` of `3dfgat_atmos` as an example, I can remove/rework it depending on how we want to use this task. Files are now linked instead of copied

@mranst thanks Mike, I will pull this branch today and see how it works for me. I'll provide feedback...

mranst committed:
Merge branch 'develop' into feature/mranst/get_geos_jedi_background

mranst committed:
Update naming and add task to 3dvar_atmos

mranst committed:
add task

mranst committed:
Add task to other suites

rtodling commented:
@mranst This now works for me just fine. Very nice!

mranst committed:
Merge branch 'develop' into feature/mranst/get_geos_jedi_background

mranst commented:
> @mranst This now works for me just fine. Very nice!

Great! If you want to approve the PR I can merge it

Dooruk reviewed (dismissed):
Sorry I know this is not related to me but I get email updates for swell PRs and this looks good 👍🏼

mranst committed:
add runtime for getobsnotinr2d2

mranst commented:
> Sorry I know this is not related to me but I get email updates for swell PRs and this looks good 👍🏼

Caught a mistake, sorry. Could I get it again? Thanks
